does not compile on FreeBSD 10.3-STABLE / related to pull #2316

It does not compile here on

10.3-STABLE FreeBSD 10.3-STABLE #0 r320135: Tue Jun 20 13:08:24 CEST 2017

Error msg is (tried building it with a fresh checkout):

cc -DHAVE_CONFIG_H -I. -I..  -DVARLIB_DIR="\"/opt/netdata/var/lib/netdata\""  -DCACHE_DIR="\"/opt/netdata/var/cache/netdata\""  -DCONFIG_DIR="\"/opt/netdata/etc/netdata\""  -DLOG_DIR="\"/opt/netdata/var/log/netdata\""  -DPLUGINS_DIR="\"/opt/netdata/usr/libexec/netdata/plugins.d\""  -DRUN_DIR="\"/opt/netdata/var/run/netdata\""  -DWEB_DIR="\"/opt/netdata/usr/share/netdata/web\""                 -O2 -I/usr/local/include -D_THREAD_SAFE -pthread -MT freebsd_ipfw.o -MD -MP -MF .deps/freebsd_ipfw.Tpo -c -o freebsd_ipfw.o freebsd_ipfw.c
freebsd_ipfw.c:30:12: error: unknown type name 'ipfw_cfg_lheader'
    static ipfw_cfg_lheader *cfg = NULL;
           ^
freebsd_ipfw.c:36:5: error: use of undeclared identifier 'ipfw_obj_ctlv'
    ipfw_obj_ctlv *ctlv = NULL;
    ^
freebsd_ipfw.c:36:20: error: use of undeclared identifier 'ctlv'
    ipfw_obj_ctlv *ctlv = NULL;
                   ^
freebsd_ipfw.c:37:5: error: use of undeclared identifier 'ipfw_obj_tlv'
    ipfw_obj_tlv *rbase = NULL;
    ^
freebsd_ipfw.c:37:19: error: use of undeclared identifier 'rbase'
    ipfw_obj_tlv *rbase = NULL;
                  ^
freebsd_ipfw.c:53:5: error: use of undeclared identifier 'ipfw_obj_tlv'
    ipfw_obj_tlv *tlv;
    ^
freebsd_ipfw.c:53:19: error: use of undeclared identifier 'tlv'
    ipfw_obj_tlv *tlv;
                  ^
freebsd_ipfw.c:89:22: error: use of undeclared identifier 'IPFW_CFG_GET_STATIC'
        cfg->flags = IPFW_CFG_GET_STATIC | IPFW_CFG_GET_COUNTERS | IPFW_CFG_GET_STATES;
                     ^
freebsd_ipfw.c:89:44: error: use of undeclared identifier 'IPFW_CFG_GET_COUNTERS'
        cfg->flags = IPFW_CFG_GET_STATIC | IPFW_CFG_GET_COUNTERS | IPFW_CFG_GET_STATES;
                                           ^
freebsd_ipfw.c:89:68: error: use of undeclared identifier 'IPFW_CFG_GET_STATES'
        cfg->flags = IPFW_CFG_GET_STATIC | IPFW_CFG_GET_COUNTERS | IPFW_CFG_GET_STATES;
                                                                   ^
freebsd_ipfw.c:91:23: error: use of undeclared identifier 'IP_FW_XGET'
        op3->opcode = IP_FW_XGET;
                      ^
freebsd_ipfw.c:107:26: error: use of undeclared identifier 'IPFW_CFG_GET_STATIC'
            cfg->flags = IPFW_CFG_GET_STATIC | IPFW_CFG_GET_COUNTERS | IPFW_CFG_GET_STATES;
                         ^
freebsd_ipfw.c:107:48: error: use of undeclared identifier 'IPFW_CFG_GET_COUNTERS'
            cfg->flags = IPFW_CFG_GET_STATIC | IPFW_CFG_GET_COUNTERS | IPFW_CFG_GET_STATES;
                                               ^
freebsd_ipfw.c:107:72: error: use of undeclared identifier 'IPFW_CFG_GET_STATES'
            cfg->flags = IPFW_CFG_GET_STATIC | IPFW_CFG_GET_COUNTERS | IPFW_CFG_GET_STATES;
                                                                       ^
freebsd_ipfw.c:109:27: error: use of undeclared identifier 'IP_FW_XGET'
            op3->opcode = IP_FW_XGET;
                          ^
freebsd_ipfw.c:120:9: error: use of undeclared identifier 'ctlv'
        ctlv = (ipfw_obj_ctlv *) (cfg + 1);
        ^
freebsd_ipfw.c:120:17: error: use of undeclared identifier 'ipfw_obj_ctlv'
        ctlv = (ipfw_obj_ctlv *) (cfg + 1);
                ^
freebsd_ipfw.c:120:32: error: expected expression
        ctlv = (ipfw_obj_ctlv *) (cfg + 1);
                               ^
freebsd_ipfw.c:122:26: error: use of undeclared identifier 'IPFW_CFG_GET_STATIC'
        if (cfg->flags & IPFW_CFG_GET_STATIC) {
                         ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
*** [freebsd_ipfw.o] Error code 1

make[2]: stopped in /root/netdata/src
1 error

make[2]: stopped in /root/netdata/src
*** [all-recursive] Error code 1

make[1]: stopped in /root/netdata
1 error

make[1]: stopped in /root/netdata
*** [all] Error code 2

make: stopped in /root/netdata
1 error

make: stopped in /root/netdata
 FAILED

--- a/src/freebsd_ipfw.c
+++ b/src/freebsd_ipfw.c
@@ -15,6 +15,8 @@
 
 int do_ipfw(int update_every, usec_t dt) {
     (void)dt;
+#if __FreeBSD__ >= 11
+
     static int do_static = -1, do_dynamic = -1, do_mem = -1;
 
     if (unlikely(do_static == -1)) {
@@ -350,4 +352,9 @@ int do_ipfw(int update_every, usec_t dt) {
     }
 
     return 0;
+#else
+    error("FREEBSD: ipfw charts supported for FreeBSD 11.0 and newer releases only");
+    COMMON_IPFW_ERROR();
+    return 1;
+#endif
 }

